main.go

```diff
@@ -146,7 +147,7 @@ func main() {
 		}
 
 		// remove trailing whitespace from each html line for markdown renderers
-		s := regexp.MustCompile(`(?m)^\s+`).ReplaceAllString(b.String(), "")
```

Comment:
This regex matches leading whitespace, not trailing whitespace. AFAICT, this defeats attempts to use `<pre>` tags in doc comments, which are super helpful for providing YAML examples of CRD usage. @ahmetb would you take a PR that just adjusts the regex?

Reply:
It was my intention to remove the LEADING whitespace here. the // comment is incorrect.

Output of this program is a markdown file (with HTML syntax). For that to work on github properly, we have to unindent all HTML lines.